subsys/bluetooth/host/smp.c

```diff
@@ -292,25 +300,34 @@ static bool le_sc_supported(void)
 	       BT_CMD_TEST(bt_dev.supported_commands, 34, 2);
 }
 
-static uint8_t get_io_capa(void)
+static const struct bt_conn_auth_cb *get_auth_cb(struct bt_smp *smp)
+{
+	atomic_ptr_cas(&smp->auth_cb, BT_SMP_AUTH_CB_UNINITIALIZED, (atomic_ptr_val_t)bt_auth);
+
+	return atomic_ptr_get(&smp->auth_cb);
+}
```

Comment:
IMO a `get` function should be stateless and not modify the input. 

If you want to keep this in a single function, please rename to something like `get_or_set`

Reply:
To ensure that callbacks' access would be safe this needs to be wrapped in one function. I will rename it to `latch_auth_cb` (because that's what the function actually does)

Reply:
Sure, latch/bind would also work :)